[2019-10-04 12:59:30,567] {taskinstance.py:620} INFO - Dependencies all met for <TaskInstance: capstone_project_dag.Load_i94_immigration_staging_table 2019-10-04T17:53:03.266837+00:00 [queued]>
[2019-10-04 12:59:30,589] {taskinstance.py:620} INFO - Dependencies all met for <TaskInstance: capstone_project_dag.Load_i94_immigration_staging_table 2019-10-04T17:53:03.266837+00:00 [queued]>
[2019-10-04 12:59:30,590] {taskinstance.py:838} INFO - 
--------------------------------------------------------------------------------
[2019-10-04 12:59:30,590] {taskinstance.py:839} INFO - Starting attempt 2 of 4
[2019-10-04 12:59:30,590] {taskinstance.py:840} INFO - 
--------------------------------------------------------------------------------
[2019-10-04 12:59:30,598] {taskinstance.py:859} INFO - Executing <Task(CopyS3ToRedshiftOperator): Load_i94_immigration_staging_table> on 2019-10-04T17:53:03.266837+00:00
[2019-10-04 12:59:30,598] {base_task_runner.py:133} INFO - Running: ['airflow', 'run', 'capstone_project_dag', 'Load_i94_immigration_staging_table', '2019-10-04T17:53:03.266837+00:00', '--job_id', '599', '--pool', 'default_pool', '--raw', '-sd', 'DAGS_FOLDER/project_dag.py', '--cfg_path', '/var/folders/dr/9yvjlnm11xv25k2gknf2n3z00000gn/T/tmpd8babauq']
[2019-10-04 12:59:31,247] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table /usr/local/lib/python3.7/site-packages/airflow/configuration.py:627: DeprecationWarning: You have two airflow.cfg files: /Users/victorialemay/airflow/airflow.cfg and /Users/victorialemay/Courses/Udacity/Data Engineering Nanodegree/capstone/project-archive/airflow/airflow.cfg. Airflow used to look at ~/airflow/airflow.cfg, even when AIRFLOW_HOME was set to a different value. Airflow will now only read /Users/victorialemay/Courses/Udacity/Data Engineering Nanodegree/capstone/project-archive/airflow/airflow.cfg, and you should remove the other file
[2019-10-04 12:59:31,248] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table   category=DeprecationWarning,
[2019-10-04 12:59:31,396] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table [2019-10-04 12:59:31,396] {settings.py:213} INFO - settings.configure_orm(): Using pool settings. pool_size=5, max_overflow=10, pool_recycle=1800, pid=81992
[2019-10-04 12:59:31,744] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table [2019-10-04 12:59:31,743] {__init__.py:51} INFO - Using executor LocalExecutor
[2019-10-04 12:59:32,232] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table [2019-10-04 12:59:32,231] {dagbag.py:90} INFO - Filling up the DagBag from /Users/victorialemay/Courses/Udacity/Data Engineering Nanodegree/capstone/project-archive/airflow/dags/project_dag.py
[2019-10-04 12:59:32,277] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table [2019-10-04 12:59:32,277] {cli.py:516} INFO - Running <TaskInstance: capstone_project_dag.Load_i94_immigration_staging_table 2019-10-04T17:53:03.266837+00:00 [running]> on host 1.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.ip6.arpa
[2019-10-04 12:59:32,323] {copy_s3_to_redshift.py:46} INFO - Using data file  from S3 bucket project-capstone-parquet
[2019-10-04 12:59:32,323] {copy_s3_to_redshift.py:47} INFO - Inserting into destination Redshift table i94_immigration_staging
[2019-10-04 12:59:32,370] {logging_mixin.py:95} INFO - [[34m2019-10-04 12:59:32,369[0m] {[34mbase_hook.py:[0m84} INFO[0m - Using connection to: [1mid: redshift. Host: redshift-cluster-1.c4i0rfzboubh.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: {}[0m[0m
[2019-10-04 12:59:32,939] {logging_mixin.py:95} INFO - [[34m2019-10-04 12:59:32,939[0m] {[34mdbapi_hook.py:[0m171} INFO[0m - TRUNCATE TABLE i94_immigration_staging[0m
[2019-10-04 12:59:33,273] {copy_s3_to_redshift.py:56} INFO - Copying data from S3 to Redshift
[2019-10-04 12:59:33,289] {logging_mixin.py:95} INFO - [[34m2019-10-04 12:59:33,289[0m] {[34mbase_hook.py:[0m84} INFO[0m - Using connection to: [1mid: redshift. Host: redshift-cluster-1.c4i0rfzboubh.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: {}[0m[0m
[2019-10-04 12:59:33,885] {logging_mixin.py:95} INFO - [[34m2019-10-04 12:59:33,885[0m] {[34mdbapi_hook.py:[0m171} INFO[0m - 
        COPY i94_immigration_staging
        FROM 's3://project-capstone-parquet/'
        IAM_ROLE 'arn:aws:iam::191711490296:role/myRedshiftRole'
        FORMAT AS PARQUET
    [0m
[2019-10-04 12:59:36,657] {taskinstance.py:1051} ERROR - S3 Query Exception (Fetch)
DETAIL:  
  -----------------------------------------------
  error:  S3 Query Exception (Fetch)
  code:      15001
  context:   Task failed due to an internal error. Unmatched number of columns between table and file. Table columns: 29, Data columns: 28, File name: https://s3.us-west-2.amazonaws.com/project-capstone-parquet/part-00000-b9542815-7a8d-45fc-9c67-c9c5007ad0d4-c000.snapp
  query:     1005
  location:  dory_util.cpp:931
  process:   query0_121_1005 [pid=21747]
  -----------------------------------------------

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 926, in _run_raw_task
    result = task_copy.execute(context=context)
  File "/Users/victorialemay/Courses/Udacity/Data Engineering Nanodegree/capstone/project-archive/airflow/plugins/operators/copy_s3_to_redshift.py", line 74, in execute
    redshift.run(formatted_sql)
  File "/usr/local/lib/python3.7/site-packages/airflow/hooks/dbapi_hook.py", line 172, in run
    cur.execute(s)
psycopg2.errors.InternalError_: S3 Query Exception (Fetch)
DETAIL:  
  -----------------------------------------------
  error:  S3 Query Exception (Fetch)
  code:      15001
  context:   Task failed due to an internal error. Unmatched number of columns between table and file. Table columns: 29, Data columns: 28, File name: https://s3.us-west-2.amazonaws.com/project-capstone-parquet/part-00000-b9542815-7a8d-45fc-9c67-c9c5007ad0d4-c000.snapp
  query:     1005
  location:  dory_util.cpp:931
  process:   query0_121_1005 [pid=21747]
  -----------------------------------------------


[2019-10-04 12:59:36,659] {taskinstance.py:1074} INFO - Marking task as UP_FOR_RETRY
[2019-10-04 12:59:36,693] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table Traceback (most recent call last):
[2019-10-04 12:59:36,693] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table   File "/usr/local/bin/airflow", line 32, in <module>
[2019-10-04 12:59:36,694] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table     args.func(args)
[2019-10-04 12:59:36,694] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table   File "/usr/local/lib/python3.7/site-packages/airflow/utils/cli.py", line 74, in wrapper
[2019-10-04 12:59:36,694] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table     return f(*args, **kwargs)
[2019-10-04 12:59:36,694] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table   File "/usr/local/lib/python3.7/site-packages/airflow/bin/cli.py", line 522, in run
[2019-10-04 12:59:36,694] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table     _run(args, dag, ti)
[2019-10-04 12:59:36,694] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table   File "/usr/local/lib/python3.7/site-packages/airflow/bin/cli.py", line 440, in _run
[2019-10-04 12:59:36,694] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table     pool=args.pool,
[2019-10-04 12:59:36,694] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table   File "/usr/local/lib/python3.7/site-packages/airflow/utils/db.py", line 74, in wrapper
[2019-10-04 12:59:36,695] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table     return func(*args, **kwargs)
[2019-10-04 12:59:36,695] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table   File "/usr/local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 926, in _run_raw_task
[2019-10-04 12:59:36,695] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table     result = task_copy.execute(context=context)
[2019-10-04 12:59:36,695] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table   File "/Users/victorialemay/Courses/Udacity/Data Engineering Nanodegree/capstone/project-archive/airflow/plugins/operators/copy_s3_to_redshift.py", line 74, in execute
[2019-10-04 12:59:36,695] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table     redshift.run(formatted_sql)
[2019-10-04 12:59:36,695] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table   File "/usr/local/lib/python3.7/site-packages/airflow/hooks/dbapi_hook.py", line 172, in run
[2019-10-04 12:59:36,695] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table     cur.execute(s)
[2019-10-04 12:59:36,695] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table psycopg2.errors.InternalError_: S3 Query Exception (Fetch)
[2019-10-04 12:59:36,695] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table DETAIL:  
[2019-10-04 12:59:36,695] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table   -----------------------------------------------
[2019-10-04 12:59:36,695] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table   error:  S3 Query Exception (Fetch)
[2019-10-04 12:59:36,695] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table   code:      15001
[2019-10-04 12:59:36,696] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table   context:   Task failed due to an internal error. Unmatched number of columns between table and file. Table columns: 29, Data columns: 28, File name: https://s3.us-west-2.amazonaws.com/project-capstone-parquet/part-00000-b9542815-7a8d-45fc-9c67-c9c5007ad0d4-c000.snapp
[2019-10-04 12:59:36,696] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table   query:     1005
[2019-10-04 12:59:36,696] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table   location:  dory_util.cpp:931
[2019-10-04 12:59:36,696] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table   process:   query0_121_1005 [pid=21747]
[2019-10-04 12:59:36,696] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table   -----------------------------------------------
[2019-10-04 12:59:36,696] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table 
[2019-10-04 12:59:36,696] {base_task_runner.py:115} INFO - Job 599: Subtask Load_i94_immigration_staging_table 
[2019-10-04 12:59:40,565] {logging_mixin.py:95} INFO - [[34m2019-10-04 12:59:40,564[0m] {[34mlocal_task_job.py:[0m105} INFO[0m - Task exited with return code 1[0m
